libs/wai-utilities/src/Network/Wai/Utilities/Server.hs

```diff
@@ -202,7 +202,9 @@ errorHandlers =
   [ Handler $ \(x :: Wai.Error) -> pure x,
     Handler $ \(_ :: InvalidRequest) -> pure $ Wai.Error status400 "client-error" "Invalid Request",
     Handler $ \(_ :: TimeoutThread) -> pure $ Wai.Error status408 "client-error" "Request Timeout",
-    Handler $ \(ZlibException (-3)) -> pure $ Wai.Error status400 "client-error" "Invalid request body compression",
+    Handler $ \case
+      ZlibException (-3) -> pure $ Wai.Error status400 "client-error" "Invalid request body compression"
+      ZlibException _ -> pure $ Wai.Error status500 "server-error" "Server Error",
```

Comment:
This is pretty much the current behaviour I think, but maybe we want something else?

Reply:
Perhaps the idea of the incomplete pattern was that it should *only* throw if the pattern matches, and the handler should not be picked otherwise.  But I think `runHandlers` below does not allow for this behavior.  If the type matches, this is the last handler that's called.

So I'm in favor of leaving this as is.